code/game/objects/items/breakable.dm

```diff
@@ -0,0 +1,129 @@
+//Breakable items
+
+//TODO:
+//multiples of a single fragment
+//add damaged text into the general examine proc
+//check slimes
+//add thrown response
+//projectiles?
+//kicking and biting?
+//change "hit" to something else?
+//hit sounds
+//punching hurting hands and stuff?
+//use kicking code
+//generalize out of devices
+
+/obj/item/device
+
+	//Destructability parameters:
+	var/breakable = 0 //1: breakable (by smashing it using the following flags
+	var/breakable_flags = 0 /*possible flags include BREAKABLE_ALL | BREAKABLE_HIT | BREAKABLE_HIT_EMPTY | BREAKABLE_HIT_WEAPON | BREAKABLE_THROW
+							BREAKABLE_HIT encompasses both BREAKABLE_HIT_EMPTY and BREAKABLE_HIT_WEAPON */
+	var/struct_integ = 15 //structural integrity of the item, akin to HP.
+	var/struct_integ_max = 15
+	var/damage_armor = 5 //attacks of this much damage or below will glance off
+	var/damage_resist = 5 //attacks stronger than damage_armor will have their damage reduced by this much
+	var/damaged_text = "" //Addendum to the description when it's damaged eg. damaged_text of "It is dented." Empty string "" will skip this addendum.
+	var/breaks_text = "" //Visible message when the items breaks. eg. "breaks apart" Empty string skips this.
+	var/breaks_sound = "" //path to audible sound when the item breaks. Empty string skips this.
+	var/list/breakable_fragments = list() //List of objects that will be produced when the item is broken apart. eg. /obj/weapon/item/shard
```

Comment:
Creating an empty list for each item is not the end of the world but also not great for memory usage. You should leave it uninitialized (which defaults to null) and only create it when it's used. The code using this will have to be made aware of this.

Additionally, I don't know if a simple list of things to spawn will be flexible enough for a system like this. I think a lot of implementations will want custom behavior to transfer fingerprints and whatnot from the item being destroyed.

Reply:
I'll take a look at fingerprint transferring.

Reply:
For now it transfers fingerprints and suit fibers, as well as blood information and generates a blood overlay if the item was bloody itself. So smashing a blood-stained beaker would result in blood-stained shards.